dungeon/src/task/quizquestion/Quiz.java

```diff
@@ -0,0 +1,149 @@
+package task.quizquestion;
+
+import task.Task;
+import task.TaskContent;
+
+/**
+ * Represents a single quiz question, including the question itself, possible answer choices, and
+ * the type of question.
+ *
+ * <p>A {@link Quiz} can be a Single-Choice, a Multiple-Choice, or a Free-text question. The type is
+ * stored as a {@link Type}. If the question is asked via the UI, the {@link QuizUI} will configure
+ * the UI for the question based on that type. The type can be accessed via {@link #type()}.
+ *
+ * <p>Add a {@link Content} answer by using the {@link #addAnswer(Content.Type, String)} method. Use
+ * the {@link #contentStream()} method to get the answers as a stream.
+ *
+ * <p>The question will be stored as {@link Content} and can be accessed via {@link #question()}.
+ */
+public class Quiz extends Task {
+
+    private final Type type;
+    private final Content question;
+
+    /**
+     * Create a new {@link Quiz} with the given configuration.
+     *
+     * <p>This will create a new {@link Content} instance as the question reference.
+     *
+     * <p>The {@link Quiz} will not have any answers, use {@link #addAnswer(Content.Type, String)}
+     * to add possible answers to the question.
+     *
+     * @param type Type of the question (e.g., single-choice)
+     * @param questionContentType What does the question contain? Just text, just a path to an
+     *     image, or both text and a path to an image?
+     * @param questionText The question itself (can contain a path to images).
+     */
+    public Quiz(final Type type, Content.Type questionContentType, String questionText) {
+        super();
+        this.type = type;
+        taskText(questionText);
+        question = new Content(this, questionContentType, questionText);
+    }
+
+    /**
+     * Get the type of the question.
+     *
+     * @return The type of the question.
+     */
+    public Type type() {
+        return type;
+    }
+
+    /**
+     * Create a {@link Content} answer instance with the given configuration as a possible answer
+     * for this question.
+     *
+     * @param type What does the answer contain? Just text, just a path to an image, or both text
+     *     and a path to an image?
+     * @param content The answer itself (can contain a path to images).
+     */
+    public void addAnswer(Content.Type type, String content) {
```

Comment:
warum übergeben wir hier nicht eine objektreferenz vom typ `content`? wäre irgendwie einfacher, als hier die konstruktoren von `content` nochmal zu spiegeln, oder?

Reply:
das binden des content-objekts an den task kann ja per methoden-aufruf in add() gemacht werden ...

Reply:
1. Ich als Aufrufer will eine Antwort zum Quiz hinzufügen, dass es sich dabei um ein `Content` handelt, ist mir eigentlich egal.
2. Ich hab als Nutzer einen einzigen Aufruf den ich machen muss `quiz.addAnswer(...)` und muss nicht erst ein Objekt erstellen (das mir danach egal ist) und bin dann auch noch dafür verantwortlich es zu registrieren. 
3. Ich vermeide Tasklose Content-Objekte. Es gibt (aktuell) keinen Grund ein Content-Objekt zu erstellen und dieses nicht direkt beim Task registrieren zu wollen. 



Man könnte die Methode noch das erzeugte Content-Objekt zurückliefern lassen, falls das irgendwie nötig ist. 

@malt-r sag du mal was, du musst es am Ende benutzen.

Reply:
> das binden des content-objekts an den task kann ja per methoden-aufruf in add() gemacht werden ...

Einfach um es mal aufzuschreiben: 
1. `Quiz.Content`Objekt mit Typ und String erzeugen
2. `Quiz#add` mit erzeugten Content-Objekt aufrufen
3. Innerhalb von `Quiz#add` noch `Content#task` mit `this` aufrufen. 

=> Scheint mir jetzt nicht unbedingt einfacher.

Reply:
von außen (von der dsl her) schon

Reply:
Nach der Besprechung mit @malt-r : So wie @cagix sagt

Reply:
ist jetzt so umgesetzt.
Zu beachten ist: Das sobald einmal der Task per `TaskContent#task(Task task)` gesetzt ist, dass dieses dann wie `final` zu verstehen ist. 
Grund dafür ist, dass der `Task` auch eine Referenz auf das `TaskComponent` speichert. Würde die Referenz geändert werden, müsste das `TaskContent` Objekt aus dem `Task` entfernt werden. 
Das könnte allerlei konzeptionelle folge Fragen aufwerfen und ich sehe aktuell wirklich keinen Grund ein TaskContent zu erstellen, einem Task zuzordnen und dann einen anderen Task zuzordnen. 

@malte-r meinung?